alchemist/alchemist-influence-sphere/src/main/kotlin/it/unibo/alchemist/sensory/FieldOfView2D.kt

```diff
@@ -0,0 +1,16 @@
+package it.unibo.alchemist.sensory
+
+import it.unibo.alchemist.model.interfaces.Position2D
+import it.unibo.alchemist.shapes.GeometricShape2D
+import java.awt.geom.Arc2D
+
+// To be better implemented following https://legends2k.github.io/2d-fov/design.html
+class FieldOfView2D<P : Position2D<P>>(
+    originX: Double,
+    originY: Double,
+    direction: Double,
+    aperture: Double,
+    distance: Double
+) : InfluenceSphere2D<P>(GeometricShape2D<P>(Arc2D.Double(originX - distance, originY - distance,
```

Comment:
Format this class better, all that left whitespace is useless

Reply:
You mean to go new line at 14? White spaces are correct according to ktlintCheck opinion

Reply:
now I undestand which whitespaces you are talking about

Reply:
Ktlint helps, but cannot understand when reading is good. This is bad:
```kotlin
) : InfluenceSphere2D<P>(GeometricShape2D<P>(Arc2D.Double(originX - distance, originY - distance,
                                                            distance * 2, distance * 2,
                                                            -direction, -aperture, Arc2D.PIE)))
```
as there is a lot of wasted space on the left hand side. What is that for? Why indenting so many times?

```kotlin
) : InfluenceSphere2D<P>(GeometricShape2D<P>(Arc2D.Double(
    originX - distance, originY - distance,
    distance * 2, distance * 2,
    -direction, -aperture, Arc2D.PIE
)))
```
Much better, isn't it?

But any of the following works:
```kotlin
) : InfluenceSphere2D<P>(
        GeometricShape2D<P>(
            Arc2D.Double(originX - distance, originY - distance, distance * 2, distance * 2, -direction, -aperture, Arc2D.PIE)
       )
)
```
```kotlin
) : InfluenceSphere2D<P>(GeometricShape2D<P>(Arc2D.Double(
    originX - distance,
    originY - distance,
    distance * 2,
    distance * 2,
    -direction,
    -aperture,
    Arc2D.PIE
)))
```
```kotlin
) : InfluenceSphere2D<P>(GeometricShape2D<P>(
    Arc2D.Double(originX - distance, originY - distance, distance * 2, distance * 2, -direction, -aperture, Arc2D.PIE)
))
```
(the latter probably the clearest)

Anyway, pick the one you prefer, just don't leave so much empty columns on the left. It's unreadable otherwise.

Reply:
Perfect, I've already modified it like the last you suggested me